corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js

```diff
@@ -25,4 +25,73 @@ describe('Render a case list', function () {
             assert.isTrue(caseTileGridView instanceof MenuList.GridCaseTileListView);
         });
     });
-});
\ No newline at end of file
+
+    describe('#getMenus', function() {
+        var server,
+            clock,
+            requests;
+
+        beforeEach(function() {
+            window.gettext = sinon.spy();
+
+            FormplayerFrontend.regions = {
+                loadingProgress: {
+                    currentView: null,
+                    empty: sinon.spy(),
+                    show: sinon.spy(),
+                }
+            };
+
+            requests = [];
+            clock = sinon.useFakeTimers();
+            server = sinon.useFakeXMLHttpRequest();
+            server.onCreate = function (xhr) {
+                requests.push(xhr);
+            };
+
+        });
+
+        afterEach(function() {
+            server.restore();
+            clock.restore()
+        });
+
+        it('Should execute an async restore', function() {
```

Comment:
this test is a bit on the longer side, but i think does a good job of giving a higher level overview of how this works